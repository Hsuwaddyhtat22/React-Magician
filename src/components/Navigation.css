.navigation {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 5px 10px;
  background-color: #c9a61a;
}

.logo a {
  text-decoration: none;
  color: rgb(98, 21, 21);
  font-weight: bold;
  font-size: 35px;
}

.nav-links {
  list-style: none;
  display: flex;
}

.nav-links li {
  margin-right: 20px;
}

.nav-links a {
  text-decoration: none;
  color: rgb(98, 21, 21);
  font-weight: bold;
  font-size: 22px;
}

.active-link {
  color: #f90;
}
