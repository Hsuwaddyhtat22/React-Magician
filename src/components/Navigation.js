import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import './Navigation.css';

function Navigation() {
  return (
    <nav className="navigation">
      {' '}
      { ' ' }
      {' '}
      {}
      {' '}
      { ' ' }
      {' '}
      <div className="logo">
        <Link to="/">
          {' '}
          { ' ' }
          MATH MAGICIANS
          {' '}
          { ' ' }
          {' '}

        </Link>
      </div>

      { /* Navigation links on the right */ }
      {' '}
      { ' ' }
      {' '}
      <ul className="nav-links">
        <li>
          <NavLink
            to="/"
            exact
            activeClassName="active-link"
          >
            Home
            {' '}
            { ' ' }
          </NavLink>
        </li>
        {' '}
        { ' ' }
        {' '}
        <li>
          <NavLink
            to="/calculator"
            activeClassName="active-link"
          >
            Calculator
            {' '}
            { ' ' }
          </NavLink>
        </li>
        {' '}
        { ' ' }
        {' '}
        <li>
          <NavLink
            to="/quote"
            activeClassName="active-link"
          >
            Quotes
            {' '}
            { ' ' }
          </NavLink>
        </li>
        {' '}
        { ' ' }

      </ul>
      {' '}
      { ' ' }
      {' '}

    </nav>
  );
}

export default Navigation;
