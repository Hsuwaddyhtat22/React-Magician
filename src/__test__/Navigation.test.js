import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from '../components/Navigation';

test('renders the Navigation component', () => {
    render( <
        Router >
        <
        Navigation / >
        <
        /Router>
    );

    // Check if the navigation logo is rendered
    expect(screen.getByText('MATH MAGICIANS')).toBeInTheDocument();

    // Check if the "Home" link is present
    expect(screen.getByText('Home')).toBeInTheDocument();

    // Check if the "Calculator" link is present
    expect(screen.getByText('Calculator')).toBeInTheDocument();

    // Check if the "Quotes" link is present
    expect(screen.getByText('Quotes')).toBeInTheDocument();
});

test('activates the correct navigation link when on that page', () => {
    render( <
        Router >
        <
        Navigation / >
        <
        /Router>
    );

    // Check if the "Home" link is active when on the home page
    expect(screen.getByText('Home').closest('a')).toHaveClass('active');

    // Check if the "Calculator" link is active when on the calculator page
    // You can simulate this by changing the route or providing a custom route
    // that matches the Calculator page.
    // For example, you can use something like <Router initialEntries={['/calculator']}>.
    // Then, you can expect the "Calculator" link to have the "active" class.

    // Check if the "Quotes" link is active when on the quotes page
    // Similar to the "Calculator" link, you can simulate this by changing the route
    // or providing a custom route that matches the Quotes page.

    // Adjust the test cases accordingly for the 'Calculator' and 'Quotes' links.
});