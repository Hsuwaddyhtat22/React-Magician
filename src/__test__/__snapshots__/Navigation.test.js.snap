// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Navigation component 1`] = `
<div>
  <nav
    class="navigation"
  >
     
     
     
     
     
     
    <div
      class="logo"
    >
      <a
        href="/"
      >
         
         
        MATH MAGICIANS
         
         
         
      </a>
    </div>
     
     
     
    <ul
      class="nav-links"
    >
      <li>
        <a
          activeclassname="active-link"
          aria-current="page"
          class="active"
          href="/"
        >
          Home
           
           
        </a>
      </li>
       
       
       
      <li>
        <a
          activeclassname="active-link"
          class=""
          href="/calculator"
        >
          Calculator
           
           
        </a>
      </li>
       
       
       
      <li>
        <a
          activeclassname="active-link"
          class=""
          href="/quote"
        >
          Quotes
           
           
        </a>
      </li>
       
       
    </ul>
     
     
     
  </nav>
</div>
`;
